src/adc.rs

```diff
@@ -449,6 +449,311 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
+                    // I'd log a warning but the log crate is not available here
+                    None
+                } else {
+                    Some(cal_handle)
+                }
+            }
+            #[cfg(not(any(
+                esp32c3,
+                all(
+                    esp32c6,
+                    not(all(esp_idf_version_major = "5", esp_idf_version_minor = "0")),
+                    not(esp_idf_version_full = "5.1.0")
+                ),
+                esp32s3,
+            )))]
+            None
+        }
+
+        #[allow(unused_variables)]
+        fn get_line_calibration_handle(
+            unit_id: u8,
+            atten: adc_atten_t,
+            bitwidth: adc_bits_width_t,
+        ) -> Option<adc_cali_handle_t> {
+            #[cfg(any(esp32, esp32c2, esp32s2))]
+            {
+                // esp32 has an additional field that the exanple defalts
+                // to using fuse values for vref. Maybe we should expose
+                // this as a config option?
+                #[allow(clippy::needless_update)]
+                let cal_config = adc_cali_line_fitting_config_t {
+                    unit_id: unit_id as u32,
+                    atten,
+                    bitwidth,
+                    ..Default::default()
+                };
+                let mut cal_handle: adc_cali_handle_t = core::ptr::null_mut();
+                if let Err(_err) = unsafe {
+                    esp!(esp_idf_sys::adc_cali_create_scheme_line_fitting(
+                        &cal_config,
+                        &mut cal_handle
+                    ))
+                } {
+                    // I'd log a warning but the log crate is not available here
+                    None
+                } else {
+                    Some(cal_handle)
+                }
+            }
+            #[cfg(not(any(esp32, esp32c2, esp32s2)))]
+            None
+        }
+
+        fn pin(&mut self) -> &mut PeripheralRef<'d, T> {
+            &mut self.pin
+        }
+
+        #[inline(always)]
+        pub fn read(&mut self) -> Result<u16, EspError> {
+            let raw = self.read_raw()?;
+            self.raw_to_cal(raw)
+        }
+
+        #[inline(always)]
+        pub fn read_raw(&mut self) -> Result<u16, EspError> {
+            let channel = self.pin().adc_channel();
+            self.adc.borrow().read_raw_internal(channel)
+        }
+
+        #[inline(always)]
+        pub fn raw_to_cal(&self, raw: u16) -> Result<u16, EspError> {
+            if let Some(calibration) = &self.calibration {
+                self.adc.borrow().raw_to_cal_internal(*calibration, raw)
+            } else {
+                Ok(raw)
+            }
+        }
+    }
+
+    unsafe impl<'d, T, ADC: Adc, M> Send for AdcChannelDriver<'d, T, ADC, M>
+    where
+        T: ADCPin<Adc = ADC>,
+        M: Borrow<AdcDriver<'d, ADC>>,
+    {
+    }
+
+    unsafe impl<'d, T, ADC: Adc, M> Sync for AdcChannelDriver<'d, T, ADC, M>
```

Comment:
`Sync` is not necessary. Remove.

Reply:
remove from `AdcDriver` as well?

Reply:
No. As per my earlier comment, you *need* `AdcDriver` to be *both* `Send` and `Sync`, **only** because it is used - **as a reference** - in `AdcChannelDriver`. (The `M: Borrow` thing). If it was "owned" by `AdcChannelDriver`, then it would have been enough for `AdcDriver` to be just `Send` - as AdcChannelDriver. But no - because of the `Borrow` thing - it is either owned, or used by reference. And because of the latter, it needs to additionally be `Sync`. In other words, `Send` for a structure implies that - if it contains references - these need to be `Sync`.